src/core/column/cut.h

```diff
@@ -197,6 +196,89 @@ class Cut_ColumnImpl : public Virtual_ColumnImpl {
 };
 
 
+template <bool RIGHT_CLOSED>
+class CutBins_ColumnImpl : public Virtual_ColumnImpl {
+  private:
+    Column col_;
+    dblvec bin_edges_;
+
+  public:
+
+    CutBins_ColumnImpl(Column&& col, dblvec&& bin_edges)
+      : Virtual_ColumnImpl(col.nrows(), dt::SType::INT32),
+        col_(std::move(col)),
+        bin_edges_(std::move(bin_edges))
+    {
+      xassert(ltype_is_numeric(col_.ltype()));
```

Comment:
Add an assert that the size of `bin_edges_` is at least 2? To prevent this class from being used incorrectly

Reply:
Yeah, let's do it.